x-pack/plugins/security_solution/public/entity_analytics/api/api.ts

```diff
@@ -111,3 +114,34 @@ export const fetchAssetCriticalityPrivileges = async (): Promise<EntityAnalytics
     }
   );
 };
+
+type AssetCriticality = SnakeToCamelCase<AssetCriticalityRecord>;
+/**
+ * Create asset criticality
+ */
+export const createAssetCriticality = async (
+  params: Pick<AssetCriticality, 'idField' | 'idValue' | 'criticalityLevel'>
+): Promise<AssetCriticalityRecord> => {
+  return KibanaServices.get().http.fetch<AssetCriticalityRecord>(ASSET_CRITICALITY_URL, {
```

Comment:
We should stop using `KibanaServices` singleton, please use the _useKibana_ hook:  `const { http } = useKibana().services`

Reply:
I see.
The problem is that the `KibanaServices` singleton is initialized only once when the plugin first starts and it never gets updated, so if a service changes for some reason after starting we won't notice and we'll keep using the initial version of the service incorrectly. 
Using the `useKibana` hook, however, takes the services from the `KibanaContextProvider` which is integrated with the React lifecycle, so we will always use the latest updated services.

I don't know if the `http` service can get updated, probably not. But I think it would be better to forget about this potential problem and use the hook as a general rule of thumb.
But that's ultimately up to you.

Reply:
oh! That's useful to know!
Yeah, I'm definitely thinking it's best to tackle this later. See, the thing is these functions are called as part of the `react-query` query functions, so we can't really even use hooks here. 
I didn't think about it too much with your original comment so it didnt even occur to me, but this change implies moving quite a few things around, so it's definitely looking like out of scope. 
I'm tracking the ticket tho :)

Reply:
Yeah, it can be done separately. 
What we usually do is pass the `http` service by parameter to those functions. Or you could also have an API client **class** that receives the `http` service in the constructor and then expose a bunch of public API methods. Just some ideas